src/redux/material/material.sagas.js

```diff
@@ -85,6 +117,18 @@ function* handleAddMaterial({ payload }) {
     yield call(handleMaterialError, error);
   }
 }
+function* handleAddMaterialColor({ payload }) {
+  try {
+    yield put(setMaterialLoading(true));
+    yield call(createMaterialColor, payload);
+    yield put(clearColors());
+    yield put(setMaterialLoading(false));
+    yield call(handleSuccessSnackbar, SUCCESS_ADD_STATUS);
+    yield put(getMaterialColors(payload.id));
```

Comment:
I guess you can get your added `color` from your mutation request and then just add it to redux state, it reduce useless request to the server.

Reply:
Fixed

Reply:
I can't do this because I don't have a new color right now I have to await for it (which I can't perfome inside saga)
In other components of our project, we use  redirect, which also provides new request to the server.